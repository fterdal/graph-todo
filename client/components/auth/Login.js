import React, { Component } from 'react';
import { graphql, compose } from 'react-apollo';
import gql from 'graphql-tag';

class Login extends Component {

  state = {
    email: '',
    password: '',
  }

  login = async () => {
    const { email, password } = this.state
    await this.props.loginMutation({
      variables: {
        email,
        password
      },
      update: (store, { data }) => {
        console.log('store', store);
        console.log('data', data);
        // const data = store.readQuery({ query: ALL_LINKS_QUERY })
        // data.allLinks.splice(0, 0, createLink)
        // store.writeQuery({
        //   query: ALL_LINKS_QUERY,
        //   data
        // })
      }
    })
    this.state = {email: '', password: ''};
  }
  render() {
    // console.log('this.props', this.props);
    return (
      <form>
        <fieldset>
          <label>Email</label>
          <input
            id="emailField"
            type="text"
            value={this.state.email}
            onChange={(evt) => this.setState({ email: evt.target.value })}
          />
          <label>Password</label>
          <input
            id="passwordField"
            type="password"
            value={this.state.password}
            onChange={(evt) => this.setState({ password: evt.target.value })}
          />
          <a
            className="button auth-button"
            onClick={() => this.login()}>
              Login
          </a>
        </fieldset>
      </form>
    )
  }
}

const loginMutation = gql`
  mutation LoginMutation($email: String!, $password: String!){
    login(
      input: {
        email: $email,
        password: $password,
      }
    ) {
      id
      email
    }
  }
`;

export default compose(
  graphql(loginMutation, {
    // name: 'loginMutation',
    props: ({ data, mutate }) => ({data, loginMutation: mutate}),
  }),
)(Login)
// export default graphql(loginMutation, { name: 'loginMutation' })(Login)
