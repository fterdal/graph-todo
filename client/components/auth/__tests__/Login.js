import React from 'react';
import renderer from 'react-test-renderer';
import { Login } from '../Login';

test('Login renders a form to log in', () => {
  const login = renderer.create(
    <Login />
  );
  let loginTree = login.toJSON();
  expect(loginTree).toMatchSnapshot();
});

test('User can enter in the login fields', () => {
  const login = renderer.create(
    <Login />
  );
  login.find('form').simulate('submit', { preventDefault: jest.fn() })
  let loginTree = login.toJSON();
  console.log('loginTree:', loginTree);
  expect(loginTree).toMatchSnapshot();
});
