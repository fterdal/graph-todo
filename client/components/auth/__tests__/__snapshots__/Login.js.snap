// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders a form to log in 1`] = `
<div
  className="container"
>
  <h1>
    Login
  </h1>
  <form>
    <fieldset>
      <label>
        Email
      </label>
      <input
        id="emailField"
        onChange={[Function]}
        type="text"
        value=""
      />
      <label>
        Password
      </label>
      <input
        id="passwordField"
        onChange={[Function]}
        type="password"
        value=""
      />
      <a
        className="button auth-button"
        onClick={[Function]}
      >
        Login
      </a>
    </fieldset>
  </form>
</div>
`;

exports[`User can enter in the login fields 1`] = `
<div
  className="container"
>
  <h1>
    Login
  </h1>
  <form>
    <fieldset>
      <label>
        Email
      </label>
      <input
        id="emailField"
        onChange={[Function]}
        type="text"
        value=""
      />
      <label>
        Password
      </label>
      <input
        id="passwordField"
        onChange={[Function]}
        type="password"
        value=""
      />
      <a
        className="button auth-button"
        onClick={[Function]}
      >
        Login
      </a>
    </fieldset>
  </form>
</div>
`;
